usaspending_api/references/v2/views/city.py

```diff
@@ -122,21 +127,39 @@ def create_es_search(scope, search_text, country=None, state=None):
         query["must"].append({"match": {"{scope}_state_code".format(scope=scope): state}})
         query["should"] = [build_country_match(scope, "USA"), build_country_match(scope, "UNITED STATES")]
         query["minimum_should_match"] = 1
-    elif country == "FOREIGN":
-        # Create a "Should Not" query with a nested bool, to get everything non-USA
-        query["should"] = {
-            "bool": {
-                "must_not": [build_country_match(scope, "USA"), build_country_match(scope, "UNITED STATES")]
-            }
-        }
     elif country != "USA":
         # A non-USA selected country
-        query["must"].append({"match": {"{scope}_country_code".format(scope=scope): country}})
+        if country != "FOREIGN":
+            query["must"].append({"match": {"{scope}_country_code".format(scope=scope): country}})
+        # Create a "Should Not" query with a nested bool, to get everything non-USA & non-null
+        query["should"] = [
+            {
+              "bool": {
+                "must": {
+                  "exists": {
+                    "field": "{}_country_code".format(scope)
+                  }
+                },
+              }
+            }
+          ]
+        query["should"][0]["bool"]["must_not"] = [build_country_match(scope, "USA"),
+                                                  build_country_match(scope, "UNITED STATES")]
+        query["minimum_should_match"] = 1
```

Comment:
This is a duplicate with the code in location_code_geofilter. We should move this to a helper function to DRY up the code.